docs/design/additional-endpoints.md

```diff
@@ -0,0 +1,299 @@
+# Block Explorer API Support
+
+## Overview
+
+This document outlines the design for implementing new package for REST API endpoints allowing exposure of more EVM centric data that support Block Explorer (e.g. Etherscan, BlockScout) API needs.
+
+The endpoints will support:
+*  ERC20 token transfers 
+*  ERC721 NFT transfers
+* ERC1155 multi-token transfers (TBD)
+*  Tokens owned by an address (TBD)
+
+## Problem Statement
+
+Currently MN doesn't support EVM centric queries, like token transfer events for specific standards like ERC20, ERC721, ERC1155 or balance of an address for a specific token.
+
+## Goals
+*  Provide API endpoints to fetch token transfer events
+*  Support filtering by sender address
+*  Support filtering by receiver address
+*  Support filtering by block range
+*  Support filtering by specific token contract
+*  Support ERC20, ERC721, and ERC1155 token standards
+*  Match Etherscan's response format for easy integration
+*  Maintain performance with large datasets
+*  Provide endpoints to fetch tokens owned by an address
+
+## Non-Goals
+*  Support all Blockscout and Etherscan API endpoints
+*  Support other events except for Transfer events
+
+## Proposed Solution
+
+Introduce a new package `rest-api` with the following REST endpoints:
+
+| Endpoint | Description 
+|----------|-------------|
+| `GET /api/token/transfers?standard={standard}&address={address}&contractaddress={contractAddress}&startblock={blockNum}&endblock={blockNum}&page={page}&offset={offset}&sort={asc\|desc}` | Fetch token transfer events 
+| `GET /api/account/{address}/tokens?page={page}&offset={offset}` | Fetch tokens owned by an address
+
+The package will be a standalone package with no relay dependencies, its own mirror node client implementation and its own cache service conncting to redis.
+This allows for a more modular and scalable solution, with the ability to easily add more endpoints in the future.
+
+
+https://api.etherscan.io/api?module=account&action=tokentx&contractaddress=0x9f8f72aa9304c8b593d555f12ef6589cc3a579a2&address=0x4e83362442b8d1bec281594cea3050c8eb01311c&page=1&offset=100&startblock=0&endblock=27025780&sort=asc&apikey=MW96VKBCE6WV6TXWK87E4Q12K6BQ5AG9PS
+### Package Structure
+```
+ packages/
+   rest-api/
```

Comment:
Hmm, another approach would be `rest-server` package with the controllers under there and the services under the `relay` package or this new `rest-api` package.
In this way the service implementation could be change in the future a transparent way or the underlying server logic - both packages decoupled from each other.
Let's chat about it, the above is food for thought and might hint towards the final form we want to get to in a tentative refactor of the repo